### PR TITLE
WIP: feat: Resolve anchor origin recursively using origin property

### PR DESCRIPTION
**Pull request by DRK3**
Signed-off-by: Derek Trider <Derek.Trider@securekey.com>

### CODE REVIEW

File: pkg/anchor/writer/writer.go [hunk 1]
```diff
@@ -558,39 +565,108 @@ func (c *Writer) getWitnesses(refs []*operation.Reference) ([]string, error) {
...
+				finalAnchorOrigin, anchorOriginIPNSAddressWithScheme, ipnsAddressChain)
+
+			break
+		} else {
+			logger.Debugf("Resolved anchor origin IPNS address %s from %s, but this doesn't match the original "+
+				"anchor origin IPNS address (%s). %s will be resolved next if the max number of resolve levels "+
+				"hasn't been hit yet. Current IPNS address chain: %v.",
+				newAnchorOriginIPNSAddress, currentAnchorOriginIPNSAddressWithScheme,
+				anchorOriginIPNSAddressWithScheme, newAnchorOriginIPNSAddress, ipnsAddressChain)
+			currentAnchorOriginIPNSAddressWithScheme = newAnchorOriginIPNSAddress
```
Comment (fqutishat) [inline pkg/anchor/writer/writer.go:617]: you need to handle if origin return not ipns. it's possible HTTPS

Comment (fqutishat) [reply pkg/anchor/writer/writer.go:617]: @DRK3 https://github.com/hyperledger/aries-framework-go-ext/pull/132#discussion_r652164974

File: pkg/anchor/writer/writer.go [hunk 2]
```diff
@@ -558,39 +565,108 @@ func (c *Writer) getWitnesses(refs []*operation.Reference) ([]string, error) {
...
+			originEndpointForSuffix, err)
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return "", fmt.Errorf("received status code %d from origin endpoint %s",
+			resp.StatusCode, originEndpointForSuffix)
 	}
 
-	logger.Debugf("Successfully resolved anchor origin %s from %s", witnessPropertyString, anchorOrigin)
+	err = resp.Body.Close()
```
Comment (bstasyszyn) [inline pkg/anchor/writer/writer.go:664]: Response body needs to be closed regardless of whether status not OK. You should add a defer close right before ioutil.ReadAll.

Comment (DRK3) [reply pkg/anchor/writer/writer.go]: Oops, good catch. Fixed!

**Pull request closed without merging**